compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -2319,6 +2325,71 @@ generateDirectCall(TR::CodeGenerator * cg, TR::Node * callNode, bool myself, TR:
 
       AOTcgDiag1(comp, "\ntempInst=%p\n", tempInst);
       return tempInst;
+#elif defined(TR_TARGET_64BIT) && defined(J9ZOS390) 
+      // If trampolines are enabled we will see if we need to use one
+      if (comp->getOption(TR_EnableTrampolines))
+         {
+         // With MultiCodeCache, we ALWAYS can jump to at least a trampoline.
+         #if !defined(PUBLIC_BUILD)
```

Comment:
Currently trampolines are always enabled on zLinux64, trampolines are not needed in zLinux32 and zOS32. So for the case of zOS64 we want it to go through this path.
I will delete the "#if !definied(PUBLIC_BUILD)", it used to be there from previous code and has now been removed.